docs/content/querying/groupbyquery.md

```diff
@@ -220,7 +220,8 @@ When using the "v2" strategy, the following runtime properties apply:
 |`druid.query.groupBy.maxOnDiskStorage`|Maximum amount of disk space to use, per-query, for spilling result sets to disk when either the merging buffer or the dictionary fills up. Queries that exceed this limit will fail. Set to zero to disable disk spilling.|0 (disabled)|
 |`druid.query.groupBy.singleThreaded`|Merge results using a single thread.|false|
 |`druid.query.groupBy.forceHashAggregation`|Force to use hash-based aggregation.|false|
-|`druid.query.groupBy.intermediateCombineDegree`|The number of intermediate nodes combined together in the combining tree. Higher degrees will need a less numer of threads which might be helpful to improve the query performance by reducing the overhead of too many threads if the server has sufficiently powerful cpu cores.|8|
+|`druid.query.groupBy.intermediateCombineDegree`|Number of intermediate nodes combined together in the combining tree. Higher degrees will need a less numer of threads which might be helpful to improve the query performance by reducing the overhead of too many threads if the server has sufficiently powerful cpu cores.|8|
+|`druid.query.groupBy.numParallelCombineThreads`|Hint for the number of parallel combining threads. This should be larger than 1 to turn on the parallel combining feature. The actual number of threads used for parallel combining is min(`druid.query.groupBy.numParallelCombineThreads`, `druid.processing.numThreads`).|1|
```

Comment:
Those configs are impossible to get right for an average user. Either they should be simpler, or descriptions should be much more elaborate, explaining in what conditions user would want to change the values, and to what particular values

Reply:
For this patch, I think more elaborate descriptions are the way to go, especially something that describes when a user might want to change it from the default.

Additionally, since we have a lot of groupBy settings now, I'd like to organize them into "commonly tuned configs" and "advanced configs, not commonly changed". IMO the commonly tuned configs would be:

- druid.processing.numMergeBuffers (tuned to trade off concurrent query capability vs. memory footprint)
- druid.query.groupBy.maxMergingDictionarySize (if you do large groupBys, this should be tuned to fit well with druid.processing.numMergeBuffers and your JVM heap setting. if you don't do large groupBys, it doesn't matter much)
- druid.query.groupBy.maxOnDiskStorage (if you do large groupBys, this should be tuned to fit your needs to use disk for aggregation)

Hopefully, the rest of the parameters are not commonly tuned.

Reply:
I agree

Reply:
Sounds good. I'll improve the document.

Reply:
I added new sections 'Performance tuning for groupBy v2', 'commonly tuned configurations', and 'advanced configurations'.
You can see in a rendered form here.
- https://gist.github.com/jihoonson/efdd3f847695533c2d3cc251b9eb703d#performance-tuning-for-groupby-v2
 - https://gist.github.com/jihoonson/efdd3f847695533c2d3cc251b9eb703d#configurations